lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -55,6 +55,10 @@ function isVersionPointer(
   return hasKey('ref', obj);
 }
 
+function normalizeVersionPointer(versionPointer: string): string {
+  return versionPointer?.replace(regEx(/[._]/g), '-');
```

Comment:
```suggestion
  return versionPointer.replace(regEx(/[._]/g), '-');
```

`versionPointer` can't be optional here

Reply:
fixed